File: CPP/PROGRAM/Peripheral.cpp
#pragma once
#include <iostream>
#include <string>
#include "Komponen.cpp"

using namespace std;

class Peripheral : public Komponen{
    //atribut
    private:
        string tipePeripheral;
        string interface;

    public:
        //konstruktor
        Peripheral(){
            this->tipePeripheral = "";
            this->interface = "";
        }

        Peripheral(string tipePeripheral, string interface, string merk, string nama) : Komponen(merk, nama){
            this->tipePeripheral = tipePeripheral;
            this->interface = interface;
        }

        //getter dan setter
        void setTipePeripheral(string tipePeripheral){
            this->tipePeripheral = tipePeripheral;
        }

        void setInterface(string interface){
            this->interface = interface;
        }

        string getTipePeripheral(){
            return this->tipePeripheral;
        }

        string getInterface(){
            return this->interface;
        }

        ~Peripheral(){

        }
};

File: PYTHON/PROGRAM/Peripheral.py
from Komponen import Komponen

class Peripheral(Komponen):
    #konstruktor
    def __init__(self, merk = "", nama = "", tipePeripheral = "", interface = ""):
        super().__init__(merk, nama)
        self.__tipePeripheral = tipePeripheral
        self.__interface = interface

    #getter dan setter
    def getTipePeripheral(self):
        return self.__tipePeripheral
    
    def setTipePeripheral(self, tipePeripheral):
        self.__tipePeripheral = tipePeripheral

    def getInterface(self):
        return self.__interface
    
    def setInterface(self, interface):
        self.__interface = interface